File: 1 - INICIANTE/1047.cpp
#include <iostream>
#include <iomanip>
#include <cmath>
using namespace std;

int main() {
  int ho,hf,mo,mf,ht,mt,a,b,c,x,y;

  cin >> ho >> mo >> hf >> mf;

  ht = (hf-ho);
  mt = (mf-mo);

   // 7 8 6 7

  if ((ho==hf)&&(mf==mo)) {
    cout << "O JOGO DUROU 24 HORA(S) E 0 MINUTO(S)" <<"\n";
    }

   else if (((hf-ho)==1) && (mo>mf)) {
    ht = 0;
    a = 60-mo;
    b = a + mf;
    cout << "O JOGO DUROU " << ht << " HORA(S) " << "E " << b << " MINUTO(S)" << "\n";
  }

  else if ((ho<hf) && (mo>mf)) {
    a = hf - ho - 1;
    b = (mf - mo) + 60;
    cout << "O JOGO DUROU " << a << " HORA(S) " << "E " << b << " MINUTO(S)" << "\n";
  }

  else if ((ho==hf) && (mo<mf)) {
    a = 0; 
    b = mf-mo; 
    cout << "O JOGO DUROU " << a << " HORA(S) " << "E " << b << " MINUTO(S)" << "\n";
  }

  else if (ho>hf && mo>mf) {
    a = (24-ho) + hf;
    b = mf - mo;
    x = a-1;
    y = (60-mo)+mf;
    cout << "O JOGO DUROU " << x << " HORA(S) " << "E " << y << " MINUTO(S)" << "\n";
   
  }
  else {
    cout << "O JOGO DUROU " << ht << " HORA(S) " << "E " << mt << " MINUTO(S)" << "\n";
  }

return 0;
}

File: 1 - INICIANTE/python/1047.py
value = input().split(' ')
ho, mo, hf, mf = int(value[0]), int(value[1]), int(value[2]), int(value[3])
hora, minuto = int(), int()

if hf==ho:
    if mf==mo:
        hora = 24
        minuto = 0
    elif mf>mo:
        hora = 0
        minuto = mf-mo
    elif mf<mo:
        hora = 23
        minuto = (60-mo)+mf

elif hf>ho:
    if mf==mo:
        hora = hf-ho
        minuto = 0
    elif mf>mo:
        minuto = mf-mo
        hora = hf-ho
    elif mf<mo:
        if hf-ho==1:
            hora = 0
            minuto = (60-mo)+mf
        else:
            hora = hf-ho-1
            minuto = (60-mo)+mf

elif hf<ho:
    if mf==mo:
        hora = (24-ho)+hf
        minuto = 0
    elif mf>mo:
        hora = (24-ho)+hf
        minuto = mf-mo
    elif mf<mo:
        minuto = (60-mo)+mf
        hora = (23-ho)+hf

print('O JOGO DUROU {} HORA(S) E {} MINUTO(S)'.format(hora, minuto))
